src/odemis/util/dataio.py

```diff
@@ -234,3 +239,93 @@ def splitext(path):
 
     root = path[:len(path) - len(ext)]
     return root, ext
+
+
+def add_acq_type_md(das):
+    """
+    Add acquisition type to das.
+    returns: das with updated metadata
+    """
+    streams = data_to_static_streams(das)
+    for da, stream in zip(das, streams):
+        if isinstance(stream, StaticSEMStream):
+            da.metadata[model.MD_ACQ_TYPE] = model.MD_AT_EM
+        elif isinstance(stream, StaticCLStream):
+            da.metadata[model.MD_ACQ_TYPE] = model.MD_AT_CL
+        elif isinstance(stream, StaticARStream):
+            da.metadata[model.MD_ACQ_TYPE] = model.MD_AT_AR
+        elif isinstance(stream, StaticSpectrumStream):
+            da.metadata[model.MD_ACQ_TYPE] = model.MD_AT_SPECTRUM
+        elif isinstance(stream, StaticFluoStream):
+            da.metadata[model.MD_ACQ_TYPE] = model.MD_AT_FLUO
+        else:
+            da.metadata[model.MD_ACQ_TYPE] = "Unknown"
+            logging.warning("Unexpected stream of shape %s in input data." % da.shape)
+
+    # If AR Stream is present, multiple data arrays are created. The data_to_static_streams
+    # function returns a single ARStream, so in this case many data arrays will not be assigned a
+    # stream and therefore also don't have an acquisition type.
+    for da in das:
+        if model.MD_ACQ_TYPE not in da.metadata:
+            da.metadata[model.MD_ACQ_TYPE] = model.MD_AT_AR
+    return das
+
+
+def open_files_and_stitch(infns: list, registration_method: int = REGISTER_IDENTITY, weaving_method: int = WEAVER_MEAN) -> list:
```

Comment:
Please add a test case in util/test/dataio_test.py . I'd suggest the code automatically creates fake tiles as TIFF files, and then calls `open_files_and_stitch()` and check the output is a single DataArray.